### PR TITLE
HPCC-27155 Add Python3 support to vcpkg

### PR DESCRIPTION
**Pull request by GordonSmith**
Signed-off-by: Gordon Smith <GordonJSmith@gmail.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [x] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: cmake_modules/vcpkgSetup.cmake [hunk 1]
```diff
@@ -15,7 +15,9 @@ set (USE_APR OFF)
 set (USE_AZURE OFF)
 set (USE_AWS OFF)
 set (WSSQL_SERVICE OFF)
-set (USE_PYTHON3 OFF)
+set (USE_PYTHON3 ON)
+find_package(Python3 3.6 REQUIRED COMPONENTS Interpreter Development)
```
Comment (richardkchapman) [inline cmake_modules/vcpkgSetup.cmake:19]: Is 3.6 a minimum version here?

Comment (GordonSmith) [reply cmake_modules/vcpkgSetup.cmake:19]: I think so - I don't see any diff with or without it.

Comment (GordonSmith) [reply cmake_modules/vcpkgSetup.cmake:19]: FYI Still fixing the ubuntu build issue.

Comment (GordonSmith) [reply cmake_modules/vcpkgSetup.cmake:19]: BTW that line comes from the pyembed CMakeLists.txt

File: plugins/py3embed/CMakeLists.txt [hunk 2]
```diff
@@ -71,6 +71,10 @@ target_link_libraries(
     roxiemem
     jlib)
 
+if (LINUX)
```
Comment (richardkchapman) [inline plugins/py3embed/CMakeLists.txt:74]: What is this for?

**Review by richardkchapman — CHANGES_REQUESTED** [review]

File: plugins/py3embed/CMakeLists.txt (hunk 2 above)
Comment (GordonSmith) [reply plugins/py3embed/CMakeLists.txt:74]: It is to resolve some import errors:
````
/usr/bin/ld: ../../vcpkg_installed/x64-linux/debug/lib/libpython3.10d.a(posixmodule.o): in function `os_openpty_impl':
/home/runner/work/vcpkg/vcpkg/buildtrees/python3/x64-linux-dbg/./../src/v3.10.1-8b5dfc5615.clean/Modules/posixmodule.c:7247: undefined reference to `openpty'
/usr/bin/ld: ../../vcpkg_installed/x64-linux/debug/lib/libpython3.10d.a(posixmodule.o): in function `os_forkpty_impl':
/home/runner/work/vcpkg/vcpkg/buildtrees/python3/x64-linux-dbg/./../src/v3.10.1-8b5dfc5615.clean/Modules/posixmodule.c:7348: undefined reference to `forkpty'
```

Comment (GordonSmith) [reply plugins/py3embed/CMakeLists.txt:74]: And looks like it didn't work on the GH build machine...

**Comment by GordonSmith** [comment]
@richardkchapman rebased

**Review by richardkchapman — CHANGES_REQUESTED** [review]

**Comment by GordonSmith** [comment]
@richardkchapman looks like the ${Python3_LIBRARIES} was pulling in some additional dependencies

**Review by richardkchapman — APPROVED** [review]

**Comment by richardkchapman** [comment]
Looks like it needs a rebase

**Pull request merged**